internal/catalogmetadata/client/client.go

```diff
@@ -31,14 +63,47 @@ func (c *Client) Bundles(ctx context.Context) ([]*catalogmetadata.Bundle, error)
 		return nil, err
 	}
 	for _, catalog := range catalogList.Items {
-		channels, err := fetchCatalogMetadata[catalogmetadata.Channel](ctx, c.cl, catalog.Name, declcfg.SchemaChannel)
+		channels := []*catalogmetadata.Channel{}
+		bundles := []*catalogmetadata.Bundle{}
+
+		// TODO: When a catalogd release containing https://github.com/operator-framework/catalogd/pull/168
+		// is made this should be updated in favor of parsing `Catalog.Status` for the
+		// appropriate URL for fetching catalog contents.
+		req, err := http.NewRequestWithContext(ctx, http.MethodGet, fmt.Sprintf(catalogdOnClusterURLTemplate, c.baseURL, catalog.Name), nil)
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("error forming request: %s", err)
 		}
 
-		bundles, err := fetchCatalogMetadata[catalogmetadata.Bundle](ctx, c.cl, catalog.Name, declcfg.SchemaBundle)
+		resp, err := http.DefaultClient.Do(req)
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("error performing request: %s", err)
+		}
+		defer resp.Body.Close()
+
+		err = declcfg.WalkMetasReader(resp.Body, func(meta *declcfg.Meta, err error) error {
+			if err != nil {
+				return err
+			}
```

Comment:
I feel like this error is going to be super generic. Should we format it something like `"more context: %s", err`?

Reply:
I tried to add more context to it in https://github.com/operator-framework/operator-controller/pull/411/commits/79cf2386f67dfe6578dcf757139dde69385c748d - let me know if this is sufficient